sdk/src/Services/DynamoDBv2/Custom/DocumentModel/DocumentBatchWrite.cs

```diff
@@ -469,15 +469,22 @@ private static int GetNumberOfWrites(BatchWriteItemRequest request)
             return nextItems;
         }
 
-#if NETSTANDARD
-        private void CallUntilCompletion(BatchWriteItemRequest request, Dictionary<string, Dictionary<Key, Document>> documentMap, AmazonDynamoDBClient client)
-#else
         private void CallUntilCompletion(BatchWriteItemRequest request, Dictionary<string, Dictionary<Key, Document>> documentMap, IAmazonDynamoDB client)
-#endif
         {
             do
             {
-                var result = client.BatchWriteItem(request);
+#if AWS_ASYNC_API
+                // Cast the IAmazonDynamoDB to the concrete client instead, so we can access the internal sync-over-async methods
```

Comment:
nit: Move the casting logic above the loop.

Reply:
Fixed in https://github.com/aws/aws-sdk-net/pull/3388/commits/aa1e3273fae0ba8826c15a91902728713e06a1d4